### PR TITLE
Feature/kaa 141

### PR DESCRIPTION
**Pull request by izeigerman**

### CODE REVIEW

**Comment by ikulikov** [comment]
+1

File: client/client-multi/client-cpp/kaa/Kaa.hpp [hunk 1]
```diff
@@ -53,6 +53,16 @@ class Kaa
      */
     static IKaaClient& getKaaClient();
 
+    /**
+     * Pauses Kaa's work flow.
```
Comment (vzhukovskyi) [inline client/client-multi/client-cpp/kaa/Kaa.hpp]: Pauses Kaa's workflow

File: client/client-multi/client-cpp/kaa/Kaa.hpp [hunk 2]
```diff
@@ -53,6 +53,16 @@ class Kaa
...
+     */
+    static void pause();
+
+    /**
+     * Resumes Kaa's work flow.
```
Comment (vzhukovskyi) [inline client/client-multi/client-cpp/kaa/Kaa.hpp]: Resumes Kaa's workflow

File: client/client-multi/client-cpp/kaa/channel/IDataChannel.hpp [hunk 3]
```diff
@@ -133,6 +133,24 @@ class IDataChannel {
      */
     virtual void setConnectivityChecker(ConnectivityCheckerPtr checker) = 0;
 
+    /**
+     * Shutdowns channel instance. All connections and threads should be terminated. Instance can no longer be used.
```
Comment (vzhukovskyi) [inline client/client-multi/client-cpp/kaa/channel/IDataChannel.hpp]: Shuts down the channel instance. All connections and threads should be terminated. The instance can no longer be used.

File: client/client-multi/client-cpp/kaa/channel/IDataChannel.hpp [hunk 4]
```diff
@@ -133,6 +133,24 @@ class IDataChannel {
...
+     *
+     */
+    virtual void shutdown() = 0;
+
+    /**
+     * Pauses channel's workflow. Channel should stop all network activity.
```
Comment (vzhukovskyi) [inline client/client-multi/client-cpp/kaa/channel/IDataChannel.hpp]: Pauses the channel's workflow. The channel should stop all network activity.

File: client/client-multi/client-cpp/kaa/channel/IDataChannel.hpp [hunk 5]
```diff
@@ -133,6 +133,24 @@ class IDataChannel {
...
+     *
+     */
+    virtual void pause() = 0;
+
+    /**
+     * Resumes channel's workflow. Channel should restore previous connection.
```
Comment (vzhukovskyi) [inline client/client-multi/client-cpp/kaa/channel/IDataChannel.hpp]: Resumes the channel's workflow. The channel should restore the previous connection.

File: client/client-multi/client-cpp/kaa/Kaa.hpp [hunk 6]
```diff
@@ -33,7 +33,7 @@ class Kaa
     /**
      * Initialize Kaa library
      */
-    static void init(int options = 0);
+    static void init(int options = KaaClient::KAA_DEFAULT_OPTIONS);
 
     /**
      * Starts Kaa's work flow.
```
Comment (vzhukovskyi) [inline client/client-multi/client-cpp/kaa/Kaa.hpp]: Starts Kaa's workflow.

File: client/client-multi/client-cpp/kaa/channel/IKaaChannelManager.hpp [hunk 7]
```diff
@@ -35,6 +35,16 @@ class IKaaChannelManager {
 public:
 
     /**
+     * Updates the manager by setting the channel to the specified transport type.
+     *
+     * @param type transport type which is going to receive updates using the specified channel.
```
Comment (vzhukovskyi) [inline client/client-multi/client-cpp/kaa/channel/IKaaChannelManager.hpp]: @param type the type of the transport which is going to receive updates using the specified channel.

File: client/client-multi/client-cpp/kaa/channel/IKaaChannelManager.hpp [hunk 8]
```diff
@@ -129,6 +148,24 @@ class IKaaChannelManager {
      */
     virtual void setConnectivityChecker(ConnectivityCheckerPtr checker) = 0;
 
+    /**
+     * Shutdowns manager and all registered channels. Instance can no longer be used.
```
Comment (vzhukovskyi) [inline client/client-multi/client-cpp/kaa/channel/IKaaChannelManager.hpp]: Shuts down the manager and all registered channels. The instance can no longer be used.

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/KaaClient.java [hunk 9]
```diff
@@ -203,9 +206,18 @@
...
     /**
-     * Retrieves Kaa log collector
+     * <p>Retrieves Kaa log collector.</p>
      *
      * @return LogCollector object
      */
     LogCollector getLogCollector();
+
+    /**
+     * <p>Retrieves default channel implementation for the specified {@link ChannelType}.</p>
```
Comment (vzhukovskyi) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/KaaClient.java]: <p>Retrieves the default channel implementation for the specified {@link ChannelType}.</p>

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaChannelManager.java [hunk 10]
```diff
@@ -97,6 +97,18 @@
 public interface KaaChannelManager {
 
     /**
+     * Updates the manager by setting the channel to specified {@link TransportType}.
+     *
+     * @param transport
+     *            transport type which is going to receive updates using the specified channel.
```
Comment (vzhukovskyi) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaChannelManager.java]: the type of the transport which is going to receive updates using the specified channel.

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaChannelManager.java [hunk 11]
```diff
@@ -97,6 +97,18 @@
...
+     * @param channel
+     *            channel to be added.
```
Comment (vzhukovskyi) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaChannelManager.java]: the channel to be added.

File: client/client-multi/client-cpp/kaa/channel/IKaaChannelManager.hpp [hunk 12]
```diff
@@ -35,6 +35,16 @@ class IKaaChannelManager {
 public:
 
     /**
+     * Updates the manager by setting the channel to the specified transport type.
+     *
+     * @param type transport type which is going to receive updates using the specified channel.
+     * @param channel channel to be added.
```
Comment (vzhukovskyi) [inline client/client-multi/client-cpp/kaa/channel/IKaaChannelManager.hpp]: @param channel the channel to be added.

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaChannelManager.java [hunk 13]
```diff
@@ -190,4 +211,22 @@
      *
      */
     void setConnectivityChecker(ConnectivityChecker checker);
+
+    /**
+     * Shutdowns manager and all registered channels. Instance can no longer be used.
```
Comment (vzhukovskyi) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaChannelManager.java]: Shuts down the manager and all registered channels. The instance can no longer be used.

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaDataChannel.java [hunk 14]
```diff
@@ -125,4 +125,22 @@
      *
      */
     Map<TransportType, ChannelDirection> getSupportedTransportTypes();
+
+    /**
+     * Shutdowns channel instance. All connections and threads should be terminated. Instance can no longer be used.
```
Comment (vzhukovskyi) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaDataChannel.java]: Shuts down the channel instance. All connections and threads should be terminated. The instance can no longer be used.

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaDataChannel.java [hunk 15]
```diff
@@ -125,4 +125,22 @@
...
+     *
+     */
+    void shutdown();
+
+    /**
+     * Pauses channel's workflow. Channel should stop all network activity.
```
Comment (vzhukovskyi) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaDataChannel.java]: Pauses the channel's workflow. The channel should stop all network activity.

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaDataChannel.java [hunk 16]
```diff
@@ -125,4 +125,22 @@
...
+     *
+     */
+    void pause();
+
+    /**
+     * Resumes channel's workflow. Channel should restore previous connection.
```
Comment (vzhukovskyi) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaDataChannel.java]: Resumes the channel's workflow. The channel should restore the previous connection.

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaChannelManager.java [hunk 17]
```diff
@@ -116,6 +128,15 @@
...
+     * Updates the manager by removing the channel from the manager.
+     *
+     * @param id the channel's id.
+     * @see KaaDataChannel
+     *
+     */
+    void removeChannel(String id);
+
+    /**
      * Retrieves the list of current channels.
```
Comment (vzhukovskyi) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaChannelManager.java:140]: Retrieves a list of the current channels.

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaChannelManager.java [hunk 18]
```diff
@@ -97,6 +97,18 @@
 public interface KaaChannelManager {
 
     /**
+     * Updates the manager by setting the channel to specified {@link TransportType}.
```
Comment (vzhukovskyi) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaChannelManager.java]: Updates the manager by setting the channel to the specified {@link TransportType}.

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaChannelManager.java [hunk 19]
```diff
@@ -125,10 +146,10 @@
     List<KaaDataChannel> getChannels();
 
     /**
-     * Retrieves the list of channels by the specific type (HTTP, HTTP_LP etc.).
+     * Retrieves a list of channels by the specific type (HTTP, HTTP_LP etc.).
      *
      * @param type type of the channel.
-     * @return the channels' list.
+     * @return a channels' list.
```
Comment (vzhukovskyi) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/KaaChannelManager.java]: @return a list of channels.

File: client/client-multi/client-cpp/kaa/channel/IKaaChannelManager.hpp [hunk 20]
```diff
@@ -46,14 +56,23 @@ class IKaaChannelManager {
...
+     *
+     * @param channel the channel to be removed.
      * @see IDataChannel
      *
      */
     virtual void removeChannel(IDataChannelPtr channel) = 0;
 
     /**
-     * Retrieves the list of current channels.
+     * Retrieves a list of current channels.
```
Comment (vzhukovskyi) [inline client/client-multi/client-cpp/kaa/channel/IKaaChannelManager.hpp:75]: Retrieves a list of the current channels.

**Comment by vzhukovskyi** [comment]
LGTM

**Pull request merged**